librubyfmt/src/format.rs

```diff
@@ -708,20 +734,34 @@ pub fn format_method_call(ps: &mut dyn ConcreteParserState, method_call: MethodC
             };
 
             if !args.is_empty() {
-                ps.breakable_of(
-                    delims,
-                    Box::new(|ps| {
-                        ps.with_formatting_context(
-                            FormattingContext::ArgsList,
-                            Box::new(|ps| {
-                                format_list_like_thing(ps, args, false);
-                                ps.emit_collapsing_newline();
-                            }),
-                        );
-                        debug!("end of format method call");
-                        ps.wind_line_if_needed_for_array();
-                    }),
-                );
+                if args_has_single_def_expression(&args) {
+                    // If we match `def ...` as the first argument, just
+                    // emit it without any delimiters.
+                    ps.emit_space();
+
+                    if let ArgsAddStarOrExpressionList::ExpressionList(mut el) = args {
+                        let expr = el.pop().expect("checked the list is not empty");
+
+                        if let Expression::Def(def_expression) = expr {
+                            format_def(ps, def_expression);
+                        }
+                    }
```

Comment:
It feels redundant to check for this structure in `args_has_single_def_expression` and then immediately check for it again to use it, but I can't really suggest a way to simplify it.

Reply:
I used to have `args_has_single_def_expression` be a method that _returned_ the `Def` expression, but I ran into Rust borrower issues and I wasn't smart enough to resolve it, so I made the tradeoff here…